fiftyone/core/dataset.py

```diff
@@ -1630,6 +1650,267 @@ def add_frame_field(
+                "frames.detections.detections.confidence",
+                mode=1
+            )
+
+            print(dataset.list_frame_rollup_fields())
+
+
+        Args:
+            frame_path: the frame field to rollup
+            rollup_path (None): the sample-level field in which to store the
+                rollup results. If ``None``, a default field name will be
+                chosen in this form: rollup_{type}_{embedded_frame_field_name}
+            include_counts (False): determines the format to populate
+                ``rollup_path`` with:
+                -   ``False``: store a list of unique observed values on each
+                    sample
+                -   ``True``: store the results in a
+                    :class:`fiftyone.core.labels.Classifications` field whose
+                    ``label`` and ``count`` attributes encode the observed
+                    values and their counts, respectively. Only applicable when
+                    ``frame_path`` contains string values
+            overwrite (True): whether to overwrite any existing rollup field
+                with the same name
+            create_index (True): whether to create a database index for the
+                rollup field so it can be quickly filtered on
+
+        Raises:
+            ValueError: If ``frame_path`` is not a valid frame field
+        """
+        frame_rollup_key = "_frame_rollup"
+        has_frames = self._has_frame_fields()
+        if not has_frames:
+            raise ValueError("Dataset does not contain videos")
+
+        inpath, is_frame_field, list_fields, _, _ = self._parse_field_name(
+            frame_path
+        )
+
+        if not is_frame_field or inpath not in self.get_frame_field_schema(
+            flat=True
+        ):
+            raise ValueError("frame_path must be a valid frame field")
+
+        # Format: rollup_[list|counts]_{embedded_frame_field}
+        if rollup_path is None:
+            rollup_path = "rollup_"
+            rollup_path += "counts_" if include_counts else "list_"
+            rollup_path += inpath.replace(".", "_")
+
+        # Delete old field if overwriting
+        _field = self.get_field(rollup_path)
+        if _field is not None:
+            if overwrite and "_frame_rollup" in _field.info:
+                self.delete_frame_rollup_field(rollup_path)
+            else:
+                raise ValueError(f"Field '{rollup_path}' already exists")
+
+        # Create sample field up front
+        info = {frame_rollup_key: inpath}
+        if include_counts:
+            self.add_sample_field(
+                rollup_path,
+                fo.EmbeddedDocumentField,
+                embedded_doc_type=fo.Classifications,
+                info=info,
+            )
+        else:
+            self.add_sample_field(rollup_path, fo.ListField, info=info)
+
+        # Optionally create mongodb index on rollup field
+        if create_index:
```

Comment:
Well the reason I added it here is so that the inserts would be done after the index was created, thinking maybe mongodb could optimize better and the additional delay would be amortized over the set_values() calls.

But now I'm thinking you're right it may be better to do it at the end and maybe we can use the non-blocking index creation param so that we don't have to wait but it's going in the background.

Reply:
Has this been benchmarked? My intuition is that declaring the index first and then setting values will be faster than the other way around